internal/storage/dbstorage/dbstorage.go

```diff
@@ -112,10 +152,114 @@ func (pg *PostgresDB) SetURLBatch(u map[string]string) (map[string]string, error
 	return result, possibleError
 }
 
+func (pg *PostgresDB) SetUserURLBatch(u map[string]string, userID int) (map[string]string, error) {
+	result := make(map[string]string)
+	tx, err := pg.db.Begin()
+	if err != nil {
+		tx.Rollback()
+	}
+	queryCheck := "SELECT short_url FROM URLS WHERE original_url=$1 LIMIT 1 FOR UPDATE"
+	query := "INSERT INTO URLS (short_url, original_url, user_id) VALUES ($1, $2, $3)"
+	var possibleError error
+	for s := range u {
+		var keyExist string
+		errBlankKey := tx.QueryRowContext(pg.ctx, queryCheck, u[s]).Scan(&keyExist)
+		if errors.Is(errBlankKey, sql.ErrNoRows) {
+			tx.QueryRowContext(pg.ctx, query, s, u[s], userID)
+			result[s] = u[s]
+		} else {
+			possibleError = internalerrors.ErrOriginalURLAlreadyExists
+			result[keyExist] = u[s]
+		}
+	}
+	err = tx.Commit()
+	if err != nil {
+		err := tx.Rollback()
+		if err != nil {
+			return nil, err
+		}
+		return nil, fmt.Errorf("failed to commit transaction: %w", err)
+	}
+	return result, possibleError
+}
+
 func (pg *PostgresDB) Ping() error {
 	err := pg.db.Ping()
 	if err != nil {
 		return fmt.Errorf("failed to ping PostgreSQL: %w", err)
 	}
 	return nil
 }
+
+func (pg *PostgresDB) CreateUser(ctx context.Context) (int, error) {
+	//https://stackoverflow.com/questions/55923755/postgres-insert-without-any-values-for-columns-all-are-default
+	query := "INSERT INTO USERS DEFAULT VALUES RETURNING id;"
+	var userID int
+	err := pg.db.QueryRowContext(ctx, query).Scan(&userID)
+	if err != nil {
+		return 0, errors.New("postgres create userId")
```

Comment:
```suggestion
		return 0, fmt.Errorf("postgres create userId: %w", err)
```
не забываем оборачивать ошибки. Хотя про уместность сохранения user id в отдельной таблицы в данном проекте я уже писал выше.

Reply:
убрано